src/entities/tradeMining.ts

```diff
@@ -0,0 +1,155 @@
+import { providers, Contract, BigNumber as BN } from 'ethers';
+import { PendleTradeMiningQuerySet } from '@pendle/subgraph-sdk';
+import { NetworkInfo } from '../networks';
+import { distributeConstantsByNetwork, getCurrentTimestamp } from '../helpers';
+import { TokenAmount } from './tokenAmount';
+import { Token } from './token';
+
+export type Valuation = {
+  currency: String; // 'USD'
+  amount: String | Number; // '10.43'
+};
+
+export type TradeMiningUser = {
+  wallet: String; // wallet address
+  tradedVolume: Valuation;
+};
+
+type DefaultTradeMiningQUery = {
+  house: string;
+  phase: string;
+};
+
+type TradeMiningQuery = {
+  house: string;
+  phase: string;
+  numberOfTraders: number;
+};
+
+type GetUserRankQuery = {
+  house: string;
+  phase: string;
+  walletAddress: string;
+};
+
+type GetUserRankResponse = {
+  rank: number;
+  tradedVolume: Valuation;
+};
+
+export class TradeMining {
+  public methods(
+    signer: providers.JsonRpcSigner,
+    chainId?: number
+  ): Record<string, any> {
+    const networkInfo: NetworkInfo = distributeConstantsByNetwork(chainId);
+    const pendleSubgraphSdk = new PendleTradeMiningQuerySet(chainId || 43114);
```

Comment:
If the chainId is undefined, line 45 will distribute the mainnet constants, but line 46 will create a pendleSubgraphSdk instance based on Avax, will this be an issue?

Reply:
I've set it so that it will default to avax if no chainId, but I guess ideally whoever is working with the SDK, should always pass in the chain ID